test/test_dependencies.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pathlib import Path
+import platform
+
+import docker
+import pytest
+
+from ros_cross_compile.dependencies import DEPENDENCY_SCRIPT_SUBPATH
+from ros_cross_compile.dependencies import gather_rosdeps
+from ros_cross_compile.docker_client import DockerClient
+from ros_cross_compile.platform import Platform
+
+NO_MAC_REASON = 'CI environment cannot install Docker on Mac OS hosts.'
+
+
+def is_mac():
+    return platform.system() == 'Darwin'
+
+
+def make_pkg_xml(contents):
+    return """<package format="3">
+  <name>dummy</name>
+  <version>0.0.0</version>
+  <description>Test package</description>
+  <maintainer email="example@example.com">Example</maintainer>
+  <license>None</license>
+  {}
+</package>
+""".format(contents)
+
+
+RCLCPP_PKG_XML = make_pkg_xml("""
+  <buildtool_depend>ament_cmake</buildtool_depend>
+  <depend>rclcpp</depend>
+""")
+
+CUSTOM_KEY_PKG_XML = make_pkg_xml("""
+  <depend>definitely_does_not_exist</depend>
+""")
+
+
+@pytest.mark.skipif(is_mac(), reason=NO_MAC_REASON)
+def test_dummy_ros2_pkg(tmpdir):
+    ws = Path(str(tmpdir))
+    pkg_xml = ws / 'src' / 'dummy' / 'package.xml'
+    out_script = ws / DEPENDENCY_SCRIPT_SUBPATH
+    pkg_xml.parent.mkdir(parents=True)
+    pkg_xml.write_text(RCLCPP_PKG_XML)
+
+    client = DockerClient()
+    platform = Platform(arch='aarch64', os_name='ubuntu', ros_distro='dashing')
```

Comment:
I think we should have a TODO comment in this test to switch to native arch if supported